### PR TITLE
Owensmallwood/pubdash get public dashboard definition

### PR DESCRIPTION
**Pull request by owensmallwood**
**What this does**
You can use a public dashboard url to load a public dashboard. If you have a pubdash uid of `-ycskU9nz` you should be able to go to `/public-dashboards/-ycskU9nz` and it will load the source dashboard in the public dashboard view.

**Whats missing**
All panels still fetch their data from the original endpoint. This will be changed in another card so that panels will fetch their data from a query endpoint specifically for public dashboard queries.

**How to test**

1. Save your dashboard as public
2. Get your public dashboard uid from the database
3. On your local instance, go to `/public-dashboards/<YourPubDashUID`
4. Your dashboard should be loaded in a public view

Screen Shot 2022-06-06 at 12 22 11 PM

### CODE REVIEW

**Review by jalevin — APPROVED** [review]
Looks great! Nice work

File: pkg/api/dashboard_public.go [hunk 1]
```diff
@@ -16,7 +17,26 @@ func (hs *HTTPServer) GetPublicDashboard(c *models.ReqContext) response.Response
...
+		CanSave:   false,
+		CanEdit:   false,
+		CanAdmin:  false,
+		CanDelete: false,
+		Created:   dash.Created,
+		Updated:   dash.Updated,
+		Version:   dash.Version,
+		IsFolder:  false,
+		FolderId:  dash.FolderId,
+		IsPublic:  dash.IsPublic,
```
Comment (torkelo) [inline pkg/api/dashboard_public.go:34]: @owensmallwood should this not always be just set to true?

File: public/app/core/services/backend_srv.ts [hunk 2]
```diff
@@ -429,6 +429,10 @@ export class BackendSrv implements BackendService {
     return this.get<DashboardDTO>(`/api/dashboards/uid/${uid}`);
   }
 
+  getPublicDashboardByUid(uid: string) {
+    return this.get<DashboardDTO>(`/api/public/dashboards/${uid}`);
+  }
```
Comment (torkelo) [inline public/app/core/services/backend_srv.ts:434]: you don't need this, you can just use backendSrv.get()  directly

**Review by torkelo — COMMENTED** [review]
looks good!

**Pull request merged**